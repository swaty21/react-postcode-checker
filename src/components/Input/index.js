import React from 'react';

const InputBox = ({ id, onChange, value }) => {
  return <input data-testid={id} onChange={onChange} value={value} />;
};

export default InputBox;
