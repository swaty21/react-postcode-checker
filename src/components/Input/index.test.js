/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Input from './index';

const props = {
  id: 'submit',
  onChange: jest.fn(),
  value: 'G41'
};
describe('Input', () => {
  it('should render Input Component', () => {
    const { container } = render(<Input {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('should call onChange', () => {
    const { getByTestId } = render(<Input {...props} />);
    const inputElement = getByTestId('submit');
    fireEvent.change(inputElement, { target: { value: 'G41DL' } });
    expect(props.onChange).toHaveBeenCalled();
  });
});
