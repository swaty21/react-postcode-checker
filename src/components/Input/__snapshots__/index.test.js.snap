// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input Component 1`] = `
<div>
  <input
    data-testid="submit"
    value="G41"
  />
</div>
`;
